patches/api/0351-Poi-API.patch

```diff
@@ -0,0 +1,190 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Fri, 24 Dec 2021 23:24:03 -0500
+Subject: [PATCH] Poi API
+
+
+diff --git a/src/main/java/io/papermc/paper/world/poi/PoiManager.java b/src/main/java/io/papermc/paper/world/poi/PoiManager.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..d3a4972d375569abda103d82ed2d6e1842ca7d3d
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/world/poi/PoiManager.java
+@@ -0,0 +1,47 @@
++package io.papermc.paper.world.poi;
++
++import org.bukkit.Chunk;
++import org.bukkit.Location;
++import org.bukkit.block.data.BlockData;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++import java.util.Collection;
++import java.util.Set;
++import java.util.function.Predicate;
++
++public interface PoiManager {
++
++    @Nullable
++    PoiType getPoiType(BlockData blockData);
++
++    @NotNull
++    Set<BlockData> getBlockTypes(@NotNull PoiType poiType);
++
++    int getMaximumTickets(@NotNull PoiType poiType);
++
++    int getPathFindMinDistance(@NotNull PoiType poiType);
++
++    @NotNull
++    default Collection<RegisteredPoi> getPoisInChunk(@NotNull Location location) {
++        return getPoisInChunk(location.getBlockX() >> 4, location.getBlockZ() >> 4);
++    }
++
++    @NotNull
++    default Collection<RegisteredPoi> getPoisInChunk(@NotNull Chunk chunk) {
++        return getPoisInChunk(chunk.getX(), chunk.getZ());
++    }
++
++    @NotNull
++    Collection<RegisteredPoi> getPoisInChunk(int x, int z);
++
++    @NotNull
++    Collection<RegisteredPoi> getPoisInRange(@NotNull Location location, int range);
++
++    @Nullable
++    RegisteredPoi getClosestPoi(@NotNull Location location, int range, @NotNull Predicate<PoiType> poiTypePredicate, @NotNull Predicate<Location> locationPredicate);
```

Comment:
Maybe create the mirror function (locateNearestPOI) in bukkit world since there already locateNearestStructure and locateNearestBiome. Also why not the name PoiBlock instead of RegisteredPoi ?